common/djangoapps/util/keyword_substitution.py

```diff
@@ -46,14 +47,19 @@ def substitute_keywords(string, user_id, context):
     KEYWORD_FUNCTION_MAP = {
         '%%USER_ID%%': lambda: anonymous_id_from_user_id(user_id),
         '%%USER_FULLNAME%%': lambda: context.get('name'),
+        '%%COURSE_ID%%': lambda: context.course_id.html_id(),
```

Comment:
@bradenmacdonald: Will this break under the Blockstore runtime?

Reply:
No, [it won't break thanks to `shims.py`](https://github.com/edx/edx-platform/blob/2eeae3b9ad07fdff2cc87ae67f0f651332950443/openedx/core/djangoapps/xblock/runtime/shims.py#L349-L359), but I consider calling `block.runtime.course_id` deprecated and would prefer to see `block.scope_ids.usage_id.context_key` used instead.

Reply:
It might break in the email context though, if `context.course_id` is an AttributeError.